pkg/image-manager/controller.go

```diff
@@ -135,6 +183,9 @@ func (h *imageEntityHandler) Delete(ctx context.Context, obj entitystore.Entity)
 
 	i := obj.(*Image)
 
+	h.ForceLock(i)
```

Comment:
This seems like a race condition, if there is an `Add` operation in progress and we deleted the image in between, `h.Store.UpdateWithError` will try to update a non-existent entity, is that safe?